### PR TITLE
feat(pie-modal): added `IsFullWidthBelowMid` prop

### PR DESCRIPTION
**Pull request by LTurns**
## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [x] If it is a component change, I have reviewed the Storybook preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-modal/test/visual/pie-modal.spec.ts [hunk 1]
```diff
@@ -88,3 +91,45 @@ sizes.forEach((size) => {
...
+            isOpen: true,
+            size: 'medium',
+            isFullWidthBelowMid: true,
+        },
+    });
+
+    await percySnapshot(page, 'Modal - size = medium & isFullWidthBelowMid = true');
+});
+
+test('Should not render with full width when size is NOT medium and isFullWidthBelowMid = true', async ({ page, mount }) => {
```
Comment (xander-marjoram) [inline packages/components/pie-modal/test/visual/pie-modal.spec.ts:109]: What is the viewport width for these tests? Large modals should also be full width below the breakpoint regardless of the value of `isFullWidthBelowMid`.

Comment (xander-marjoram) [reply packages/components/pie-modal/test/visual/pie-modal.spec.ts:109]: Actually these are visual tests so the viewport varies, perhaps we should use `small` for these?

Comment (LTurns) [reply packages/components/pie-modal/test/visual/pie-modal.spec.ts:109]: yeah I wasn't sure if there was a way I could only specify a particular viewport in the test. I'll change to small - that makes more sense 👍

Comment (LTurns) [reply packages/components/pie-modal/test/visual/pie-modal.spec.ts:109]: done 😄 thank you!

**Review by kevinrodrigues — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**